src/frontend/src/handler/alter_table_column.rs

```diff
@@ -284,13 +286,6 @@ pub async fn handle_alter_table_column(
         }
     };
 
-    if columns.is_empty() {
-        Err(ErrorCode::NotSupported(
-            "alter a table with empty column definitions".to_owned(),
-            "Please recreate the table with column definitions.".to_owned(),
-        ))?
-    }
-
```

Comment:
Previously there were two cases where `columns` is empty:

1. All columns are resolved from external source instead of defined in SQL.
2. The table indeed has zero columns.

For case 1, as we always use a purified definition here, altering it is feasible. For case 2, we should have allowed it.